### PR TITLE
feat: StringList handling in default parameter processor

### PR DESCRIPTION
**Pull request by Plazmius**
## Description
Add handling for StringList parameters, which is a type of parameter, supported by AWS SSM

## Motivation and Context
This addresses https://github.com/aws/aws-dotnet-extensions-configuration/issues/48

## Testing
I have run unit and integration tests and used this code in my project

## Screenshots (if appropriate)
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

While tests are passing, I think whoever is using `DefaultParameterProcessor` won't expect that `StringList` are now handled differently. However, I would expect that one of only three SSM parameters type would convert to collection automatically.

## Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

Not sure that this feature fits the current test structure, but old tests passed

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-dotnet-extensions-configuration/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

**Comment by BradKnowles** [comment]
Thank you for this!  I used your code to create a custom `StringListParameterProcessor`, inheriting from `DefaultParameterProcessor` and it worked perfectly.

File: src/Amazon.Extensions.Configuration.SystemsManager/DefaultParameterProcessor.cs [hunk 1]
```diff
@@ -42,16 +43,33 @@ public virtual string GetKey(Parameter parameter, string path)
...
+
         public virtual IDictionary<string, string> ProcessParameters(IEnumerable<Parameter> parameters, string path)
         {
-            return parameters
-                .Where(parameter => IncludeParameter(parameter, path))
-                .Select(parameter => new
+            var result = new List<KeyValuePair<string, string>>();
+            foreach (var parameter in parameters.Where(parameter => IncludeParameter(parameter, path)))
+            {
+                if (parameter.Type == ParameterType.StringList)
```
Comment (dscpinheiro) [inline src/Amazon.Extensions.Configuration.SystemsManager/DefaultParameterProcessor.cs:61]: Thanks for opening the PR! Could you add an unit test (to `test/Amazon.Extensions.Configuration.SystemsManager.Tests/DefaultParameterProcessorTests.cs`) using a `StringList`?

Comment (Plazmius) [reply src/Amazon.Extensions.Configuration.SystemsManager/DefaultParameterProcessor.cs:61]: @dscpinheiro I have added a unit test to cover possible scenarios of usage. I tried to follow the code style of other unit tests to my best. Is there anything I can add?

Comment (dscpinheiro) [reply src/Amazon.Extensions.Configuration.SystemsManager/DefaultParameterProcessor.cs:61]: No, that was what I was looking for, thank you!

**Review by dscpinheiro — APPROVED** [review]

**Review by ashishdhingra — APPROVED** [review]
Changes work fine while adjusting and executing sample application in this repo. Sample data returned using `Get` endpoint:
```JSON
{"settings":{"stringValue":"string-value","integerValue":10,"dateTimeValue":"2000-01-01T00:00:00","booleanValue":true,"timeSpanValue":"00:05:00","stringValues":["White","Yellow","Green","Black"]}}
```

**Pull request merged**